Dynatrace/dynatrace-operator
# Introduce statefulset for OTEL collector

## Description

This PR implmenets new Reconciler: `otel.reconciler` to automate a StatefulSet for the OpenTelemetry collector for extensions.

[K8S-10311](https://dt-rnd.atlassian.net/browse/K8S-10311)

> NOTE:
>  dynatrace-extensions-collector-0     ●  0/1   CrashLoopBackOff
It's okey for now, we wait some fixes from EEC team to release new image which supports fetching config via endpoint.

## How can this be tested?

Deploy dynakube with prometheus.enabled = true

- Make sure `dynatrace-extensions-collector` statefulset is presented and has all labels/annotations.
- Also, env variables are presented for the Otelc container.

Unfortunately, you can't test more.
